File: utils/src/main/java/ai/elimu/content_provider/utils/converter/CursorToSoundGsonConverter.java
package ai.elimu.content_provider.utils.converter;

import android.database.Cursor;
import android.util.Log;

import java.util.Arrays;

import ai.elimu.model.v2.gson.content.SoundGson;

public class CursorToSoundGsonConverter {

    public static SoundGson getSoundGson(Cursor cursor) {
        Log.i(CursorToSoundGsonConverter.class.getName(), "getSoundGson");

        Log.i(CursorToSoundGsonConverter.class.getName(), "Arrays.toString(cursor.getColumnNames()): " + Arrays.toString(cursor.getColumnNames()));

        int columnIndexId = cursor.getColumnIndex("id");
        Long id = cursor.getLong(columnIndexId);
        Log.i(CursorToSoundGsonConverter.class.getName(), "id: " + id);

        int columnIndexRevisionNumber = cursor.getColumnIndex("revisionNumber");
        Integer revisionNumber = cursor.getInt(columnIndexRevisionNumber);
        Log.i(CursorToSoundGsonConverter.class.getName(), "revisionNumber: " + revisionNumber);

        int columnIndexValueIpa = cursor.getColumnIndex("valueIpa");
        String valueIpa = cursor.getString(columnIndexValueIpa);
        Log.i(CursorToSoundGsonConverter.class.getName(), "valueIpa: \"" + valueIpa + "\"");

        SoundGson soundGson = new SoundGson();
        soundGson.setId(id);
        soundGson.setRevisionNumber(revisionNumber);
        soundGson.setValueIpa(valueIpa);

        return soundGson;
    }
}

File: utils/src/main/java/ai/elimu/content_provider/utils/converter/CursorToSoundGsonConverter.kt
package ai.elimu.content_provider.utils.converter

import ai.elimu.model.v2.gson.content.SoundGson
import android.database.Cursor
import android.util.Log

object CursorToSoundGsonConverter {
    
    private const val TAG = "CursorToSoundGsonConverter"
    
    @JvmStatic
    fun getSoundGson(cursor: Cursor): SoundGson {
        Log.i(TAG, "getSoundGson")

        Log.i(TAG,
            "Arrays.toString(cursor.getColumnNames()): " + cursor.columnNames.contentToString()
        )

        val columnIndexId = cursor.getColumnIndex("id")
        val id = cursor.getLong(columnIndexId)
        Log.i(TAG, "id: $id")

        val columnIndexRevisionNumber = cursor.getColumnIndex("revisionNumber")
        val revisionNumber = cursor.getInt(columnIndexRevisionNumber)
        Log.i(TAG, "revisionNumber: $revisionNumber")

        val columnIndexValueIpa = cursor.getColumnIndex("valueIpa")
        val valueIpa = cursor.getString(columnIndexValueIpa)
        Log.i(TAG, "valueIpa: \"$valueIpa\"")

        val soundGson = SoundGson()
        soundGson.id = id
        soundGson.revisionNumber = revisionNumber
        soundGson.valueIpa = valueIpa

        return soundGson
    }
}
